### PR TITLE
Reverse sentence

### PR DESCRIPTION
**Pull request by shrutivanw**
CS Fundamentals' assignment to reverse the words in a sentence in place while preserving the count of white spaces.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,9 @@
-# reverse_sentence
+# Reverse Sentence
+In this assignment, you'll design and implement one of the string manipulation functions that is commonly asked during interviews.
+Remember that a string is an array of characters. Algorithms that worked on restricted arrays will work on strings as well.
+
+## Exercise
+* Design and implement a method to reverse the words in a sentence *in place*. For example, if the method is called with input parameter of "Yoda&nbsp;&nbsp;&nbsp;is&nbsp;&nbsp;&nbsp;&nbsp;awesome", the method should update the input string object to have the value "awesome&nbsp;&nbsp;&nbsp;&nbsp;is&nbsp;&nbsp;&nbsp;Yoda". Note that the count of white spaces between words is preserved.
+* Share and explain the time and space complexities for your solution.
+* At minimum, your implementation should pass the basic tests.
+**Note**: Do not use Ruby provided functionality for `.reverse` and `.reverse!`. You may use `.length`.
```
Comment (Hamled) [inline README.md]: We should have an extra blank line above this so that the note goes into its own paragraph.

File: lib/reverse_sentence.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# A method to reverse the words in a sentence, in place.
+def reverse_sentence(my_sentence)
+  my_sentence << "NOT IMPLEMENTED"
```
Comment (Hamled) [inline lib/reverse_sentence.rb]: See [my comment](https://github.com/AdaGold/reverse_words/pull/1/files#r160554004) on the `reverse_words` repo, which applies equally here.

File: specs/reverse_sentence_spec.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+require_relative '../lib/reverse_sentence'
+
+describe "reverse sentence" do
+  describe "basic tests" do
+    describe "test 1" do
```
Comment (Hamled) [inline specs/reverse_sentence_spec.rb]: Same as the other repos, we should probably remove the `describe` blocks that are immediate parents of individual tests.

File: specs/reverse_sentence_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        reverse_sentence(test_string)
+
+        test_string.must_equal "awesome! is Yoda"
+      end
+    end
+  end
+
+  # if it's a string parameter, check for empty
+  describe "empty string input test" do
+    it "reverse an empty sentence" do
```
Comment (Hamled) [inline specs/reverse_sentence_spec.rb]: I think this empty string test and the nil test below it could be moved into the edge cases section, and a few of those tests (such as sentence with multiple words) could be moved into the basic tests section.

**Review by Hamled — CHANGES_REQUESTED** [review]

**Comment by shrutivanw** [comment]
Thank you @Hamled PR updated to incorporate feedback.

**Review by Hamled — APPROVED** [review]
